config/config-server/src/main/resources/cruise-config.xsd

```diff
@@ -141,6 +147,38 @@
     <xsd:attribute type="nameType" name="name" use="required"/>
   </xsd:complexType>
 
+  <xsd:complexType name="secretConfigsType">
+    <xsd:sequence>
+      <xsd:element name="secretConfig" minOccurs="0" maxOccurs="unbounded" type="secretConfigType"/>
+    </xsd:sequence>
+  </xsd:complexType>
+
+  <xsd:complexType name="secretConfigType">
+    <xsd:all>
+      <xsd:element name="configuration" minOccurs="0" maxOccurs="1" type="configurationType"/>
+      <xsd:element name="rules" minOccurs="0" maxOccurs="1" type="rulesType"/>
+    </xsd:all>
+    <xsd:attribute name="id" type="nameType" use="required"/>
+    <xsd:attribute name="pluginId" type="nameType" use="required"/>
+    <xsd:attribute name="description" type="xsd:string" use="optional"/>
```

Comment:
IMO, description as an attribute seems more appropriate because it's something the SecretConfig `has` and not something it `contains`. Description can be long string of text but having it as an attribute is (more) correct.

Reply:
done